### PR TITLE
refactored

### PR DESCRIPTION
**Pull request by pnikiel**
This code needed maintenance very very badly, mostly because addNodeAndReference method has grown over time and became barely managable.
I advice to watch differences with whitespaces hidden.

### CODE REVIEW

File: src/nodemanagerbase.cpp [hunk 1]
```diff
@@ -27,6 +27,8 @@
 #include <stdexcept>
 #include <uadatavariablecache.h>
 
+static UaLocalizedText emptyDescription( "en_US", "" );
+
 NodeManagerBase::NodeManagerBase( const char* uri, bool sth, int hashtablesize ):
     m_server(0),
     m_nameSpaceUri(uri)
```
Comment (ben-farnham) [inline src/nodemanagerbase.cpp:34]: Actually refers to line 61 (I don't know how to add comments to lines that the diff doesn't show).

The refactor could also ditch BOOST_FOREACH too I guess, now we're C++11ish.

Comment (ben-farnham) [reply src/nodemanagerbase.cpp:34]: or maybe std::find_if

File: src/nodemanagerbase.cpp [hunk 2]
```diff
@@ -224,185 +226,195 @@ UA_LocalizedText make_localised( UaString text )
...
-        int inArgsSize = 0;
-        UA_Argument *outArgs = 0;
-        int outArgsSize = 0;
+    UA_Argument *inArgs = 0;
+    int inArgsSize = 0;
+    UA_Argument *outArgs = 0;
+    int outArgsSize = 0;
 
-        for ( std::list<UaNode::ReferencedTarget>::const_iterator it = referenced->cbegin(); it!=referenced->cend(); it++ )
+    for ( std::list<UaNode::ReferencedTarget>::const_iterator it = referenced->cbegin(); it!=referenced->cend(); it++ )
```
Comment (ben-farnham) [inline src/nodemanagerbase.cpp:348]: same sort of thing here re: loop iteration, could also be C++11 range based for?

**Review by ben-farnham — COMMENTED** [review]
Couple of minor things to consider whether they're worth adding in to this refactor in light of our conversion to C++11

**Comment by pnikiel** [comment]
@ben-farnham thank you, I agree with all your points. I'd prefer to open a separate ticket with your comments to keep things more managable.

**Comment by pnikiel** [comment]
@ben-farnham apart from the newly opened ticket, do you agree with the suggested changes?

**Comment by ben-farnham** [comment]
Yep - agree with the suggested changes.

**Pull request merged**